modules/install/pages/extensions.adoc

```diff
@@ -1,24 +1,316 @@
+drwxr-xr-x.  3 jboss jboss   17 Apr  9 04:13 javaee
+drwxr-xr-x. 29 jboss jboss 4096 Apr  9 04:13 javax
+drwxr-xr-x.  4 jboss jboss   35 Apr  9 04:13 net
+drwxr-xr-x. 29 jboss jboss 4096 Apr  9 04:13 org
+drwxr-xr-x.  4 jboss jboss   34 Apr  9 04:13 sun
+drwxr-xr-x.  3 jboss jboss   17 Apr  9 04:13 wildflyee
+----
 
 == Extensions
 
-== Subsystems
+* Extensions are modules that extend the core capabilities of the server.
+
+* Extensions are added by using the <extension> element, at the beginning of the JBoss EAP main configuration file: `JBOSS_HOME/standalone/configuration/standalone.xml` or `JBOSS_HOME/domain/configuration/domain.xml`.
+
+* These file defines multiple <extension> tags.
+
+* The general structure of `standalone.xml` or `domain.xml` is similar to the following content:
++
+[subs="+quotes,+macros"]
+----
+<server xmlns="urn:jboss:domain:16.0">
+    <extensions>
+	    ...list of extensions here
+    </extensions>
+
+    <system-properties>
+        ...system properties defined here
+    </system-properties>
+
+    <management>
+	    ...management interfaces defined here
+    </management>
+
+    <profile>
+	    ...list of subsystems and their configurations
+    </profile>
+
+    <interfaces>
+	    ...interface definitions
+    </interfaces>
+
+    <socket-binding-group>
+	    ...socket binding definitions
+    </socket-binding-group>
+
+    <deployments>
+	    ...deployed applications go here
+    </deployments>
+</server>
+----
+
+* For example, the following excerpt describes a list of extensions available to a server, including ejb3, which is responsible for providing the Enterprise Java Beans (EJB) runtime:
++
+[subs="+quotes,+macros"]
+----
+[vagrant@server ~]$ less /opt/EAP-8.0.0/standalone/configuration/standalone.xml
+<?xml version="1.0" encoding="UTF-8"?>
+<server xmlns="urn:jboss:domain:20.0">
+    ...
+    <extensions>
+        <!-- list all extensions that you want made available to this server -->
+        <extension module="org.jboss.as.clustering.infinispan"/>
+        <extension module="org.jboss.as.deployment-scanner"/>
+        <extension module="org.jboss.as.ejb3"/>
+        <extension module="org.jboss.as.jpa"/>
+    </extensions>
+    ...
+----
+
+* JBoss EAP uses the module name to find extension module. For example, to locate the `org.jboss.as.jpa` module, JBoss EAP searches for the `org/jboss/as/jpa/` directory in the `JBOSS_HOME/modules` directory.
+
+* The JBoss EAP core loads and activates modules/extensions only when necessary, so that inactive features do not consume memory or CPU resources unnecessarily.
+
+[NOTE]
+====
+If an organizations applications do not need a particular extension, then the system administrator can disable the extension so JBoss EAP does not load its management model anymore. Remove the <extension> entry and its corresponding <subsystem> section from the configuration file. This can be accomplished either by editing the XML directly, or using the Management CLI.
```

Comment:
we are contradicting ourselves by saying edit XML directly. Remove this sentence. always use EAP CLI or web console to manage XML config

Reply:
Removed the note